train.py

```diff
@@ -46,21 +47,50 @@ def loss_fn(y_true, y_pred):
     return tf.reduce_mean(nll)
 
 
-def crop_3d():
-    crop_size = 20
+class ReparamTrickLayer(keras.layers.Layer):
```

Comment:
What is this function doing?

Reply:
Generates a sample of oef and dbv values by using the predictions of mean and std dev for the signal generation layer?

Reply:
Found it in [the paper](https://arxiv.org/pdf/1312.6114.pdf): 2.4 The reparameterization trick